spec/consumer_spec.rb

```diff
@@ -524,4 +526,34 @@
 
     it { expect(method_original_name).to eq(:trigger_heartbeat!) }
   end
+
+  describe '#interceptor' do
+    it "creates a consumer with interceptor" do
+      interceptor = FakeConsumerInterceptor.new
+      consumer = Kafka::Consumer.new(
+        cluster: cluster,
+        logger: logger,
+        instrumenter: instrumenter,
+        group: group,
+        offset_manager: offset_manager,
+        fetcher: fetcher,
+        session_timeout: session_timeout,
+        heartbeat: heartbeat,
+        interceptors: [interceptor]
+      )
+
+      consumer.stop
+      expect(interceptor.close_count).to eq 1
+    end
+
+    it "tests on_consume chain" do
+      interceptor1 = FakeConsumerInterceptor.new('hello2', true)
+      interceptor2 = FakeConsumerInterceptor.new('hello3')
+      interceptors = Kafka::ConsumerInterceptors.new(interceptors: [interceptor1, interceptor2], logger: logger)
+      intercepted_batch = interceptors.on_consume(batch: fetched_batches[0])
+
+      expect(intercepted_batch.messages[0].value).to eq "hellohello3"
+      interceptors.close
+    end
+  end
```

Comment:
https://github.com/Shopify/ruby-kafka/pull/21#discussion_r438738041

Reply:
added tests for:
happy path where all interceptors change the value
original value if all interceptors fail

did not "if last interceptor returns error, it'll return previous values" since it's similar to "does not break the on_consume chain"